File: src/main/java/com/lcaohoanq/nocket/domain/post/PostRepository.java
package com.lcaohoanq.nocket.domain.post;

import com.lcaohoanq.nocket.domain.user.User;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface PostRepository extends JpaRepository<Post, UUID> {

    List<Post> findByUser(User user);

    List<Post> findByUserIn(List<User> users);
    
}

File: src/main/java/com/lcaohoanq/nocket/domain/post/PostRepository.kt
package com.lcaohoanq.nocket.domain.post

import com.lcaohoanq.nocket.domain.user.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface PostRepository : JpaRepository<Post?, UUID?> {
    fun findByUser(user: User): List<Post>
    fun findByUserIn(users: List<User>): List<Post>
}
